src/octree/octree_test.rs

```diff
@@ -87,29 +87,83 @@ mod tests {
         };
 
         // octree and iterator
-        let octree = build_test_octree();
+        let octree_vec: [Octree; 1] = [*build_test_octree()];
         let location = PointQuery {
             location: PointLocation::AllPoints(),
             global_from_local: None,
         };
-        let mut batch_iterator = BatchIterator::new(&octree, &location, batch_size);
+        let mut batch_iterator = BatchIterator::new(
+            &octree_vec,
+            &location,
+            batch_size,
+            std::cmp::max(1, num_cpus::get() - 1),
+            4,
+        );
 
         let _err_stop = batch_iterator
             .try_for_each_batch(callback_func)
             .expect_err("Test error");
 
-        assert_eq!(3, callback_count);
-        assert_eq!(3 * batch_size, delivered_points);
+        //  requiring and returning less points than the total
+        assert!(3 <= callback_count);
+        assert!(3 * batch_size <= delivered_points);
+        assert!(callback_count >= 3);
+        assert!((3 + 1) * batch_size >= delivered_points);
         assert!(delivered_points as i32 - max_num_points as i32 >= 0);
     }
 
+    #[test]
+    fn test_batch_iterator_more_points() {
+        let batch_size = NUM_POINTS / 2;
+        // define function
+
+        // requiring more points than in the octree
+        let mut callback_count = 0;
+        let mut delivered_points = 0;
+        let max_num_points = NUM_POINTS + 30_000;
+        println!(
+            "batch_size= {} ,  num_points= {}",
+            batch_size, max_num_points
+        );
+        let callback_func = |point_data: PointData| -> Result<()> {
+            callback_count += 1;
+            delivered_points += point_data.position.len();
+            println!(
+                "Callback_count {:}, delivered points {:}",
+                callback_count, delivered_points
+            );
+            if delivered_points >= max_num_points {
+                println!("Callback: Max Points reached!");
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Interrupted,
+                    format!("Maximum number of {} points reached.", max_num_points),
+                )
+                .into());
+            }
+            Ok(())
+        };
+        // octree and iterator
+        let octree_vec: [Octree; 1] = [*build_test_octree()];
+        let location = PointQuery {
+            location: PointLocation::AllPoints(),
+            global_from_local: None,
+        };
+
+        let mut batch_iterator = BatchIterator::new(&octree_vec, &location, batch_size, 2, 2);
+
+        let _err_stop = batch_iterator
+            .try_for_each_batch(callback_func)
+            .expect("Test OK");
+        assert!(delivered_points == NUM_POINTS + 1); // only delivers all the points in the octree
```

Comment:
It's kind of confusing that the octree has `NUM_POINTS + 1`. I would have expected `NUM_POINTS`. Can we change this?